aten/src/ATen/native/TensorShape.cpp

```diff
@@ -392,6 +390,86 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result;
+  if (tensors.size() == 0) {
+    result = at::empty({});
+    return result;
+  }
+
+  // Use at::result_type to find the right tensor to copy options from, so
+  // that the output data type is correct
+  const Tensor* options_tensor = &tensors[0];
```

Comment:
nit: use `const Tensor& options_tensor = tensor[0];`, usually in modern c++ we prefer references over pointers

Reply:
Actually, please ignore this, the code is a more complicated than I thought